File: tests/Generator/resource/rust/complex/integer_property_type.rs
use serde::{Serialize, Deserialize};
use scalar_property_type::ScalarPropertyType;

// Represents an integer value
#[derive(Serialize, Deserialize)]
pub struct IntegerPropertyType {
    #[serde(rename = "description")]
    description: Option<String>,

    #[serde(rename = "deprecated")]
    deprecated: Option<bool>,

    #[serde(rename = "type")]
    _type: Option<String>,

    #[serde(rename = "nullable")]
    nullable: Option<bool>,

}

File: tests/Generator/resource/python/complex/integer_property_type.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler, Tag
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, Annotated, Union, Literal
from .scalar_property_type import ScalarPropertyType


# Represents an integer value
class IntegerPropertyType(ScalarPropertyType):
    type: Literal["integer"] = Field(alias="type")
    pass


